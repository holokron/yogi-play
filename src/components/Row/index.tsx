import * as React from 'react'

export interface Props {
    children?: any
    className?: string
    withTopPadding?: boolean
}

export default function Row({ children, className = '', withTopPadding = false }: Props): React.ReactElement<Props> {
    if (withTopPadding) {
        className += ' pt-4'
    }

    return <div className={`row justify-content-md-center pl-3 pr-3 ${className}`}>{children}</div>
}