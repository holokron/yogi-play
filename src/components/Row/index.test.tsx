import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Row from '.'

describe('@components/Row', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Row>Test</Row>
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })

    it('should render without errors', () => {
        expect(
            shallow(
                <Row>Test</Row>
            ).matchesElement(
                <div className="row justify-content-md-center pl-3 pr-3">Test</div>
            )
        ).toBeTruthy()
    })
    
    it('should render without errors, props (withTopPadding = true)', () => {
        expect(
            shallow(
                <Row withTopPadding>Test</Row>
            ).matchesElement(
                <div className="row justify-content-md-center pl-3 pr-3 pt-4">Test</div>
            )
        ).toBeTruthy()
    })
})