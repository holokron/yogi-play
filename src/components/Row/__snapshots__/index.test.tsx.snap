// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Row renders correctly 1`] = `
<div
  className="row justify-content-md-center pl-3 pr-3 "
>
  Test
</div>
`;

exports[`@components/Row should render without errors, props (withTopPadding = true) 1`] = `
<div
  className="row justify-content-md-center pl-3 pr-3  pt-4"
>
  Test
</div>
`;
